lib/mongoose_push_web/schemas/request/send_notification/deep/mixed_notification.ex

```diff
@@ -0,0 +1,36 @@
+defmodule MongoosePushWeb.Schemas.Request.SendNotification.Deep.MixedNotification do
+  require OpenApiSpex
+  alias MongoosePushWeb.Protocols.RequestDecoder
+  alias MongoosePushWeb.Schemas.Request.SendNotification.Deep
+
+  OpenApiSpex.schema(%{
+    title: "Request.SendNotification.Deep.MixedNotification",
+    description: "In this request both alert and data fields are mandatory.",
+    type: :object,
+    properties:
+      Deep.base()[:properties]
+      |> Map.merge(Deep.alert()[:properties])
+      |> Map.merge(Deep.data()[:properties]),
+    required: Deep.base()[:required] ++ Deep.alert()[:required] ++ Deep.data()[:required],
+    example:
+      Deep.base()[:example]
+      |> Map.merge(Deep.alert()[:example])
+      |> Map.merge(Deep.data()[:example]),
+    additionalProperties: false
+  })
+
+  defimpl RequestDecoder,
+    for: MongoosePushWeb.Schemas.Request.SendNotification.Deep.MixedNotification do
+    alias MongoosePushWeb.Protocols.RequestDecoderHelper
+
+    @spec decode(%Deep.MixedNotification{}) :: MongoosePush.request()
+    def decode(schema) do
+      %{
+        service: String.to_atom(schema.service),
```

Comment:
Don't we have a helper that does this better? Even if not, then let's use `to_existing_atom` at least.

Reply:
I implemented one, following the way @michalwski already suggested [here](https://github.com/esl/MongoosePush/pull/150/#discussion_r421375560).